core/src/main/java/hudson/model/ParametersDefinitionProperty.java

```diff
@@ -69,12 +69,22 @@
     private final List<ParameterDefinition> parameterDefinitions;
 
     @DataBoundConstructor
-    public ParametersDefinitionProperty(List<ParameterDefinition> parameterDefinitions) {
+    public ParametersDefinitionProperty(@Nonnull List<ParameterDefinition> parameterDefinitions) {
+        if (parameterDefinitions == null) {
+            throw new IllegalArgumentException("ParameterDefinitions is null when this is a not valid value");
```

Comment:
Throw an a NPE if null not an IAE which is reserved for a non-null value that is not correct